src/libraries/System.Memory.Data/src/System/BinaryData.cs

```diff
@@ -21,6 +22,11 @@ public class BinaryData
     {
         private const string JsonSerializerRequiresDynamicCode = "JSON serialization and deserialization might require types that cannot be statically analyzed and might need runtime code generation.";
         private const string JsonSerializerRequiresUnreferencedCode = "JSON serialization and deserialization might require types that cannot be statically analyzed.";
+#if NETCOREAPP2_1_OR_GREATER
```

Comment:
the else block would work on any runtime, so I am not having this ifdef adds enough value.

Reply:
I agree with @KrzysztofCwalina. This should just be the else block.